File: test/python/Math/base_conversion1_to_base_10.test.py
# verification-helper: PROBLEM https://atcoder.jp/contests/abc220/tasks/abc220_b

from library.python.Math.base_conversion import to_base_10


def main():
    K = int(input())
    A, B = input().split()
    print(to_base_10([int(a) for a in A], K) * to_base_10([int(b) for b in B], K))


if __name__ == '__main__':
    main()

File: test/cpp/Math/base_conversion1_to_base_10.test.cpp
#define PROBLEM "https://atcoder.jp/contests/abc220/tasks/abc220_b"

#include "library/cpp/Math/base_conversion.cpp"
#include <iostream>

using namespace std;

int main() {
    cin.tie(nullptr);
    ios::sync_with_stdio(false);

    int K;
    string A, B;
    cin >> K >> A >> B;

    vector<int> a, b;
    for (int x: A) {
        a.emplace_back(x - '0');
    }
    for (int x: B) {
        b.emplace_back(x - '0');
    }
    cout << to_base_10(a, K) * to_base_10(b, K) << endl;

    return 0;
}